tests/integration/azure/test_credentials.py

```diff
@@ -0,0 +1,143 @@
+
+    application_id = spark_conf.get("fs.azure.account.oauth2.client.id")
+
+    end_point = spark_conf.get("fs.azure.account.oauth2.client.endpoint")
+    directory_id = end_point.split("/")[3]
+
+    secret_matched = re.search(SECRET_PATTERN, spark_conf.get("fs.azure.account.oauth2.client.secret"))
+    if secret_matched:
+        secret_scope, secret_key = (
+            secret_matched.group(1).split("/")[1],
+            secret_matched.group(1).split("/")[2],
+        )
+    assert secret_scope is not None
+    assert secret_key is not None
+
+    secret_response = ws.secrets.get_secret(secret_scope, secret_key)
+    client_secret = base64.b64decode(secret_response.value).decode("utf-8")
+
+    return MigrationTestInfo(credential_name, application_id, directory_id, secret_scope, secret_key, client_secret)
+
+
+@pytest.fixture
+def run_migration(ws, sql_backend):
+    def inner(
+        test_info: MigrationTestInfo, credentials: list[str], read_only=False
+    ) -> list[StorageCredentialValidationResult]:
+        installation = Installation(ws, 'ucx')
+        azurerm = AzureResources(ws)
+        locations = ExternalLocations(ws, sql_backend, "dont_need_a_schema")
+
+        permission_mappings = [
+            StoragePermissionMapping(
+                prefix="abfss://things@labsazurethings.dfs.core.windows.net/avoid_ext_loc_overlap",
+                client_id=test_info.application_id,
+                principal=test_info.credential_name,
+                privilege="READ_FILES" if read_only else "WRITE_FILES",
+                directory_id=test_info.directory_id,
+            )
+        ]
+        resource_permissions = StaticResourcePermissions(permission_mappings, installation, ws, azurerm, locations)
+
+        sp_infos = [
+            AzureServicePrincipalInfo(
+                application_id=test_info.application_id,
+                secret_scope=test_info.secret_scope,
+                secret_key=test_info.secret_key,
+                tenant_id="test",
+                storage_account="test",
+            )
+        ]
+        sp_crawler = StaticServicePrincipalCrawler(sp_infos, ws, sql_backend, "dont_need_a_schema")
+
+        spn_migration = StaticServicePrincipalMigration(
+            installation, ws, resource_permissions, sp_crawler, StaticStorageCredentialManager(ws, credentials)
+        )
+        return spn_migration.run(
+            MockPrompts({"Above Azure Service Principals will be migrated to UC storage credentials *": "Yes"})
+        )
+
+    return inner
+
+
+def test_spn_migration_existed_storage_credential(extract_test_info, make_storage_credential_from_spn, run_migration):
+    # create a storage credential for this test
+    make_storage_credential_from_spn(
+        credential_name=extract_test_info.credential_name,
+        application_id=extract_test_info.application_id,
+        client_secret=extract_test_info.client_secret,
+        directory_id=extract_test_info.directory_id,
```

Comment:
don't need keyword for positional arguments

Reply:
removed